tarantool_test.go

```diff
@@ -1308,6 +1429,113 @@ func TestStressSQL(t *testing.T) {
 	}
 }
 
+func TestNewPrepared(t *testing.T) {
+	// Tarantool supports SQL since version 2.0.0
+	isLess, err := test_helpers.IsTarantoolVersionLess(2, 0, 0)
+	if err != nil {
+		t.Fatalf("Could not check the Tarantool version")
+	}
+	if isLess {
+		t.Skip()
+	}
+
+	conn := test_helpers.ConnectWithValidation(t, server, opts)
+	defer conn.Close()
+
+	stmt, err := conn.NewPrepared(selectNamedQuery2)
+	if err != nil {
+		t.Errorf("failed to prepare: %v", err)
+	}
+
+	executeReq := NewExecutePreparedRequest(stmt)
+	unprepareReq := NewUnprepareRequest(stmt)
+
+	resp, err := conn.Do(executeReq.Args([]interface{}{1, "test"})).Get()
+	if err != nil {
+		t.Errorf("failed to execute prepared: %v", err)
+	}
+	if resp.Code != OkCode {
+		t.Errorf("failed to execute prepared: code %d", resp.Code)
+	}
+	if reflect.DeepEqual(resp.Data[0], []interface{}{1, "test"}) {
+		t.Error("Select with named arguments failed")
+	}
+	if resp.MetaData[0].FieldType != "unsigned" ||
+		resp.MetaData[0].FieldName != "NAME0" ||
+		resp.MetaData[1].FieldType != "string" ||
+		resp.MetaData[1].FieldName != "NAME1" {
+		t.Error("Wrong metadata")
+	}
+
+	resp, err = conn.Do(unprepareReq).Get()
+	if err != nil {
+		t.Errorf("failed to unprepare prepared statement: %v", err)
+	}
+	if resp.Code != OkCode {
+		t.Errorf("failed to unprepare prepared statement: code %d", resp.Code)
+	}
+
+	_, err = conn.Do(unprepareReq).Get()
+	if err == nil {
+		t.Errorf("the statement must be already unprepared")
+	}
+	require.Contains(t, err.Error(), "Prepared statement with id")
+
+	_, err = conn.Do(executeReq).Get()
+	if err == nil {
+		t.Errorf("the statement must be already unprepared")
+	}
+	require.Contains(t, err.Error(), "Prepared statement with id")
+
+	prepareReq := NewPrepareRequest(selectNamedQuery2)
+	resp, err = conn.Do(prepareReq).Get()
+	if err != nil {
+		t.Errorf("failed to prepare: %v", err)
+	}
+	if resp.Code != OkCode {
+		t.Errorf("failed to unprepare prepared statement: code %d", resp.Code)
+	}
+
+	stmt, err = NewPreparedFromResponse(conn, resp)
+	if err != nil {
+		t.Errorf("failed to construct a prepared object: %v", err)
+	}
+	if stmt.Conn != conn {
+		t.Errorf("the statement does not belong to the passed connection")
+	}
+
+	unprepareReq = NewUnprepareRequest(stmt)
+	resp, err = conn.Do(unprepareReq).Get()
+	if err != nil {
+		t.Errorf("failed to unprepare prepared statement: %v", err)
+	}
+	if resp.Code != OkCode {
+		t.Errorf("failed to unprepare prepared statement: code %d", resp.Code)
+	}
+}
+
+func TestNewPreparedFromResponse(t *testing.T) {
+	responses := []*Response{
+		nil,
+		{Data: nil},
+		{Data: []interface{}{"test value"}},
+		{Data: []interface{}{&Prepared{}}},
+	}
+	testConn := &Connection{}
+
+	errorsExpected := []error{
+		ErrNilResponsePassed,
+		ErrNilResponseData,
+		ErrWrongDataFormat,
+		nil,
+	}
+
+	for i, resp := range responses {
+		_, err := NewPreparedFromResponse(testConn, resp)
+		assert.Equal(t, err, errorsExpected[i])
+	}
```

Comment:
The test will stop with first fail, it better to use:

https://pkg.go.dev/testing#T.Run

See examples:
```
./decimal/decimal_test.go:140:		t.Run(testcase.numString, func(t *testing.T) {
./decimal/decimal_test.go:184:		t.Run(testcase.numString, func(t *testing.T) {
./decimal/decimal_test.go:369:		t.Run(testcase.numString, func(t *testing.T) {
./decimal/decimal_test.go:388:		t.Run(testcase.numString, func(t *testing.T) {
./decimal/decimal_test.go:415:		t.Run(testcase.numString, func(t *testing.T) {
./decimal/decimal_test.go:439:		t.Run(testcase.numString, func(t *testing.T) {
./decimal/decimal_test.go:554:		t.Run(testcase.numString, func(t *testing.T) {
./datetime/datetime_test.go:179:		t.Run(testcase.dt, func(t *testing.T) {
./datetime/datetime_test.go:199:		t.Run(tm.String(), func(t *testing.T) {
./datetime/datetime_test.go:209:		t.Run(tm.String(), func(t *testing.T) {
./datetime/datetime_test.go:519:		t.Run(testcase.dt, func(t *testing.T) {
./datetime/datetime_test.go:545:		t.Run(testcase.dt, func(t *testing.T) {
./ssl_test.go:445:			t.Run("ok_ssl_"+test.name, func(t *testing.T) {
./ssl_test.go:449:			t.Run("fail_ssl_"+test.name, func(t *testing.T) {
./ssl_test.go:457:			t.Run("ok_tnt_"+test.name, func(t *testing.T) {
./ssl_test.go:461:			t.Run("fail_tnt_"+test.name, func(t *testing.T) {
```

Reply:
Fixed.